paasta_tools/api/views/autoscaler.py

```diff
@@ -45,6 +46,25 @@ def update_autoscaler_count(request):
     service = request.swagger_data.get('service')
     instance = request.swagger_data.get('instance')
     desired_instances = request.swagger_data.get('json_body')['desired_instances']
+
+    try:
+        service_config = load_marathon_service_config(
+            service=service,
+            instance=instance,
+            cluster=settings.cluster,
+            soa_dir=settings.soa_dir,
+            load_deployments=False,
+        )
+    except:
+        error_message = 'Unable to load service config for %s.%s' % (service, instance)
+        raise ApiFailure(error_message, 404)
+
+    max_instances = service_config.get_max_instances()
+    min_instances = service_config.get_min_instances()
+    if desired_instances > max_instances or desired_instances < min_instances:
+        error_message = 'Incorrect desired_instances %d: min_instances %d max_instances %d' % (desired_instances, min_instances, max_instances)
+        raise ApiFailure(error_message, 400)
+
     set_instances_for_marathon_service(service=service, instance=instance, instance_count=desired_instances)
```

Comment:
I've been thinking about having this use `instance_count=service_config.get_instances()` to verify to the user that the changes have been logged in zookeeper. For example, if a service isn't configured for autoscaling and someone posts to its autoscaling endpoint the API will happily return {'desired_instances': foo} even though nothing actually happened.